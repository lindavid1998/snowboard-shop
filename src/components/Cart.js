import '../styles/Cart.css';

function Cart(props) {
	const { cart, isVisible, hideCart } = props;
	
	const renderedCart = cart.map((item) => <li key={item.id}>{item.name}</li>);

	return (
		<div
			className={isVisible ? 'cart' : 'cart hidden'}
			data-testid='cart-component'
		>
			<button onClick={hideCart}>Close</button>
			{cart && renderedCart}
		</div>
	);
}

export default Cart;
