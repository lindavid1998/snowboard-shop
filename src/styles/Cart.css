.cart {
  position: fixed;
  top: 0;
  right: 0;
  /* width: 300px;  */
  height: 100%; /* Cover the entire height of the viewport */
  z-index: 999; /* Ensure the element is on top of everything else */
  transition: all 0.5s;
}

.cart.hidden {
  right: -100%; 
}
