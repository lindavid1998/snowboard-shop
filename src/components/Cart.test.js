import { render, screen } from '@testing-library/react';
import Cart from './Cart';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

describe('Cart', () => {
	test('renders items, price, and checkout button if cart is not empty', async () => {
		const mockCart = [
			{
				id: 0,
				name: 'Season Kin Snowboard 2023',
				price: 349.3,
				category: 'boards',
			},
			{
				id: 1,
				name: 'GNU Young Money C2E Snowboard - Big Boys 2023',
				price: 279.99,
				category: 'boards',
			},
		];

		const cart = renderer
			.create(
				<BrowserRouter>
					<Cart cart={mockCart} />
				</BrowserRouter>
			)
			.toJSON();
		
		expect(cart).toMatchSnapshot();
	});

	test('routes to checkout page', async () => {
		const user = userEvent.setup();

		const mockCart = [
			{
				id: 0,
				name: 'Season Kin Snowboard 2023',
				price: 349.3,
				category: 'boards',
			},
			{
				id: 1,
				name: 'GNU Young Money C2E Snowboard - Big Boys 2023',
				price: 279.99,
				category: 'boards',
			},
		];
		render(
			<BrowserRouter>
				<Cart cart={mockCart} />
			</BrowserRouter>
		);

		// simulate click on checkout
		await act(async () => {
			await user.click(screen.getByText('Checkout'));
		});

		// verify routing
		expect(window.location.pathname).toBe('/checkout');
	});

	test('renders message if cart is empty', () => {
		render(
			<BrowserRouter>
				<Cart cart={[]} />
			</BrowserRouter>
		);
		expect(screen.getByText(/your cart is empty/i)).toBeTruthy()
	});
});
