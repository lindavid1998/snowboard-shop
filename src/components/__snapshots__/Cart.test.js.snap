// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders items, price, and checkout button if cart is not empty 1`] = `
<div
  className="Cart hidden"
  data-testid="cart"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-xmark fa-lg btn-close-cart"
    data-icon="xmark"
    data-prefix="fas"
    data-testid="btn-close-cart"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <li
    className="CartItem item-card"
    data-testid="cart-item"
    id={0}
  >
    <div
      className="info"
    >
      <div
        className="item-name"
        data-testid="item-name"
      >
        Season Kin Snowboard 2023
      </div>
      <div
        className="item-price"
      >
        $
        10
      </div>
      <div
        className="quantity-input"
      >
        <label
          htmlFor="quantity"
        >
          Qty:
        </label>
        <input
          data-testid="quantity"
          id="quantity"
          min="1"
          name="quantity"
          onChange={[Function]}
          type="number"
        />
      </div>
    </div>
    <img
      alt="Season Kin Snowboard 2023"
    />
    <button
      className="btn-filled btn-remove-from-cart btn-orange-text"
      onClick={[Function]}
    >
      Remove
    </button>
  </li>
  <li
    className="CartItem item-card"
    data-testid="cart-item"
    id={1}
  >
    <div
      className="info"
    >
      <div
        className="item-name"
        data-testid="item-name"
      >
        GNU Young Money C2E Snowboard - Big Boys 2023
      </div>
      <div
        className="item-price"
      >
        $
        20
      </div>
      <div
        className="quantity-input"
      >
        <label
          htmlFor="quantity"
        >
          Qty:
        </label>
        <input
          data-testid="quantity"
          id="quantity"
          min="1"
          name="quantity"
          onChange={[Function]}
          type="number"
        />
      </div>
    </div>
    <img
      alt="GNU Young Money C2E Snowboard - Big Boys 2023"
    />
    <button
      className="btn-filled btn-remove-from-cart btn-orange-text"
      onClick={[Function]}
    >
      Remove
    </button>
  </li>
  <div
    className="total-price"
    data-testid="total-price"
  >
    Total: $
  </div>
  <button
    className="btn-filled btn-checkout"
  >
    <a
      href="/checkout"
      onClick={[Function]}
    >
      Checkout
    </a>
  </button>
</div>
`;
